### DISCUSSION TITLE
Why the executable is so large?

### DISCUSSION THREAD

**Discussion by saikoml** [pocketbase/pocketbase#3665] [Q&A]
I'm new to Go but was wondering why the final executable is almost 50mb?
Is this because of the `_test.go`  files? Can they be excluded?

**Comment by ganigeorgiev** [comment]
No, while majority of the codebase is indeed tests, they are not included during regular build.

Another user also complained about this ~2 weeks ago and after investigation I found out that nearly half of the PocketBase binary size is caused by [`aws/aws-sdk-go`](https://github.com/aws/aws-sdk-go) and [`gocloud.dev/blob`](https://github.com/google/go-cloud/tree/master/blob) dependencies that are used primarily for the S3 and local filesystem abstraction. Even a minimal Go program with just the above 2 dependencies results in ~34MB unstripped binary.

There is a [task in the roadmap](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=42616173) to remove the dependencies and maybe create a thin client to send directly requests to the S3 APIs, but for now this is put on hold (_it will be done most likely after the Presentator release; see https://github.com/pocketbase/pocketbase/discussions/3271_).

**Reply by blackfyre** [reply]
Thanks for the investigation!
Those libraries are huge, independent of programming language...
